### DISCUSSION TITLE
Hosting Wikis With GitHub Pages

### DISCUSSION THREAD

**Discussion by R4356th** [community/community#22492] [Other Feature Feedback, Questions, & Ideas]
<p>Hello,<br>
I want to host wikis with GitHub Pages. Is that possible? Thank you.</p>

**Comment by R4356th** [comment]
<p>More specifically, I want to host wikis using Mediawiki.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Hello <a class="mention" href="/r4356th">@R4356th</a>, welcome to the community.</p>
<p>No, GitHub Pages is for hosting of static content only. In other words, if you want to change what’s on the website, you have to upload new HTML to the repository. Dynamically-generated content, like wikis (including Mediawiki), is not supported on GitHub Pages.</p>

**Comment by R4356th** [comment]
<aside class="quote group-github-employees" data-username="lee-dohm" data-post="4" data-topic="116096">
<div class="title">
<div class="quote-controls"></div>
 lee-dohm:</div>
<blockquote>
<p>Dynamically-generated content</p>
</blockquote>
</aside>
<p>Alright but what about forums?</p>

**Comment by lee-dohm** [comment]
<p>Forums are also dynamically-generated content.</p>
<p>Any site where you can enter stuff into the website itself to change what the website displays is dynamically generated content and is not something that is supported on GitHub Pages.</p>

**Comment by MolluscHotel** [comment]
<p>Yeah. Github is pretty upfront about not hosting dynamic pages or PHP or Node.JS or anything like that. I’m just getting into dynamic sites and I’ve opened up a whole can of worms/ learning opportunities- hosting sites, configuring Liveserver for PHP, setting up Apache, PHP vs. node.JS, etc. etc.</p>

**Reply by jxu** [reply]
Most wiki content, including text and images, are perfectly fine being statically served by GitHub Pages.

**Comment by jxu** [comment]
It's interesting some websites that function like wikis, for example MDN docs, cp-algorithms.com, osu wiki, use a git/GitHub based model of pull requests instead of wiki style anyone can edit. I think a git-based model is superior (for those who know how to make PRs) because afaik MediaWiki only supports a linear history, while having maintainers being able to comment and make edits on proposed changes is very useful.

GitHub also offers repo documentation in a limited form as repo wikis.

**Comment by plantwon1** [comment]
Wait, so no back-end coding with php!? OH NO!!!!!!!!
